Classes/EventStore/AbstractEventSourcedRepository.php

```diff
@@ -88,15 +85,9 @@ public function __construct()
     final public function save(EventSourcedAggregateRootInterface $aggregate, int $expectedVersion = null)
     {
         $streamName = $this->streamNameResolver->getStreamNameForAggregate($aggregate);
-        $uncommittedEvents = $aggregate->pullUncommittedEvents();
-        $version = $aggregate->getReconstitutionVersion() + 1;
-        $uncommittedEventsWithMetadata = array_map(function (EventInterface $event) use (&$version) {
-            return new EventWithMetadata($event, new EventMetadata([EventMetadata::VERSION => $version ++]));
```

Comment:
That was nonsense really: version should not be part of the metadata and more importantly: we can't know the version is just a sequence without gaps at this point